Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
+	}
+
+	/*
+	 *
+	 * File meta methods
+	 *
+	 */
+
+	/**
+	 * Returns an error if there is one, null shows no error.
+	 *
+	 * @return  mixed  Null for no errors, or an error array
+	 */
+	public function getError()
+	{
+		return $this->error;
+	}
+
+	/**
+	 * Returns the name of this file.
+	 *
+	 * @return  string  The name of this file
+	 */
+	public function getFilename()
+	{
+		return $this->filename;
+	}
+
+	/*
+	 *
+	 * Getter methods
+	 *
+	 */
+
+	/**
+	 * Gets the information of the file. If refresh is true,
+	 * then this method will query the wiki API again for the file details.
+	 *
+	 * @param   boolean  $refresh  True to query the wiki API again
+	 * @return  mixed              The info of the file (array), or null if error
+	 */
+	public function getInfo($refresh = false)
+	{
+		if ($refresh) { // We want to query the API
+
+			$data = array(
+				'titles' => 'File:' . $this->filename,
+				'prop' => 'info|imageinfo',
+				'iiprop' => 'bitdepth|canonicaltitle|comment|parsedcomment|'
+				          . 'commonmetadata|metadata|extmetadata|mediatype|'
+				          . 'mime|thumbmime|sha1|size|timestamp|url|user|userid',
+				'intoken' => 'edit',
+			);
+
+			$r = $this->wikimate->query($data); // Run the query
+
+			// Check for errors
+			if (isset($r['error'])) {
+				$this->error = $r['error']; // Set the error if there was one
+				return null;
+			} else {
+				$this->error = null; // Reset the error status
+			}
+
+			// Get the page (there should only be one)
+			$page = array_pop($r['query']['pages']);
+			unset($r, $data);
+
+			// Abort if invalid file title
+			if (isset($page['invalid'])) {
+				$this->invalid = true;
+				return null;
+			}
+
+			$this->edittoken = $page['edittoken'];
+
+			if (!isset($page['missing'])) {
+				// Update the existance if the file is there
+				$this->exists = true;
+				// Put the content into info
+				$this->info   = $page['imageinfo'][0];
+			}
+			unset($page);
+		}
+
+		return $this->info; // Return the info in any case
+	}
+
+	/**
+	 * Returns the aspect ratio of this image,
+	 * or 0 if file has no image dimensions.
+	 *
+	 * @return  float  The aspect ratio of this image, or 0 if no dimensions
+	 */
+	public function getAspectRatio()
+	{
+		if ($this->info['height'] > 0) {
+			return $this->info['width'] / $this->info['height'];
+		} else {
+			return 0;
+		}
+	}
+
+	/**
+	 * Returns the bit depth of this file.
+	 *
+	 * @return  int  The bit depth of this file
+	 */
+	public function getBitDepth()
+	{
+		return (int)$this->info['bitdepth'];
+	}
+
+	/**
+	 * Returns the canonical title of this file.
+	 *
+	 * @return  string  The canonical title of this file
+	 */
+	public function getCanonicalTitle()
+	{
+		return $this->info['canonicaltitle'];
+	}
+
+	/**
+	 * Returns the comment of this file.
+	 *
+	 * @return  string  The comment of this file
+	 */
+	public function getComment()
+	{
+		return $this->info['comment'];
+	}
+
+	/**
+	 * Returns the common metadata of this file.
```

Comment:
It would be nice to clarify what common (as opposed to non-common) metadata refers to.